packages/d3fc-webgl/src/series/glOhlc.js

```diff
@@ -136,24 +139,42 @@ export default () => {
     return draw;
   };
 
-  const buildBuffers = (xValues, open, high, low, close, bandwidth, lineWidth, numElements) => {
-    const buffers = {
-      xValues: [],
-      yValues: [],
-      xLineWidth: [],
-      yLineWidth: [],
-      bandwidth: [],
-      color: []
-    };
-
-    const addToBuffers = (xBuf, yBuf, xLineWidthBuf, yLineWidthBuf, bandwidthBuf) => {
-      buffers.xValues.push(xBuf);
-      buffers.yValues.push(yBuf);
-      buffers.xLineWidth.push(xLineWidthBuf);
-      buffers.yLineWidth.push(yLineWidthBuf);
-      buffers.bandwidth.push(bandwidthBuf);
-    };
+  const addToYBuffers = (values, indexPositions) => {
+    const yValBuffer = program.buffers().attribute(yValueAttrib);
+    const yDirBuffer = program.buffers().attribute(yDirectionAttrib);
+    let yArray = new Float32Array(values.length * verticesPerElement);
+    let yDirArray = new Float32Array(values.length * verticesPerElement);
+
+    if (yValBuffer) {
+      const existingYValues = yValBuffer.data();
+      yArray = yArray.map((_, i) => indexPositions.includes(i % verticesPerElement) ?
+        values[Math.floor(i / verticesPerElement)] :
+        existingYValues[i]
+      );
+      yValBuffer.data(yArray);
+
+      const existingYDirValues = yDirBuffer.data();
+      yDirArray = yDirArray.map((_, i) => indexPositions.includes(i % verticesPerElement) ?
+        yDirections[i % verticesPerElement] :
+        existingYDirValues[i]
+      );
+      yDirBuffer.data(yDirArray);
+    } else {
+      yArray = yArray.map((d, i) => indexPositions.includes(i % verticesPerElement) ?
+        values[Math.floor(i / verticesPerElement)] :
+        d
+      );
+      program.buffers().attribute(yValueAttrib, attributeBuilder(yArray).components(1));
+
+      yDirArray = yDirArray.map((d, i) => indexPositions.includes(i % verticesPerElement) ?
+        yDirections[i % verticesPerElement] :
+        d
+      );
+      program.buffers().attribute(yDirectionAttrib, attributeBuilder(yDirArray).components(1));
+    }
+  };
 
+  const setColors = (numElements) => {
```

Comment:
Could we do this in the shader? Maybe pass the colours in as uniforms?

Reply:
Each vertex requires some indication as to whether it should be green or red as it doesn't have enough information to calculate it itself. As we discussed this has now been changed to pass in an indicator of -1 (red) or 1 (green) to each vertex.